### ISSUE TITLE
Refactor several tests to reduce the time spent

### ISSUE THREAD

**Issue by atugushev** [jazzband/pip-tools#894]
#### What's the problem this feature will solve?

Currently, some tests can be refactored out to avoid using the internet to reduce the time spent.

#### Describe the solution you'd like

Refactor out below tests to use wheels from `tests/test_data/minimal_wheels`:

- test_annotate_option
- test_pre_option
- test_input_file_without_extension
- test_filter_pip_markers
- test_no_candidates
- test_no_candidates_pre
- test_stdin

#### Additional context

Following up the https://github.com/jazzband/pip-tools/issues/887#issuecomment-531540714.

**Comment by hramezani** [comment]
I can take this. is it possible for you to give me more description?

**Comment by atugushev** [comment]
Hello @hramezani! Please, feel free to take this.

Here is an example of how I would rewrite `test_stdin`:
```diff
diff --git a/tests/test_cli_compile.py b/tests/test_cli_compile.py
index 8684741..9a199d7 100644
--- a/tests/test_cli_compile.py
+++ b/tests/test_cli_compile.py
@@ -616,16 +616,15 @@ def test_not_specified_input_file(runner):
     assert out.exit_code == 2


-@pytest.mark.network
 def test_stdin(runner):
     """
     Test compile requirements from STDIN.
     """
     out = runner.invoke(
-        cli, ["-", "--output-file", "requirements.txt", "-n"], input="six==1.10.0"
+        cli, ["-", "--output-file", "requirements.txt", "-n"], input="small-fake-a==0.1"
     )

-    assert "six==1.10.0" in out.stderr
+    assert "small-fake-a==0.1" in out.stderr


 def test_multiple_input_files_without_output_file(runner):
```

The other tests could be rewritten the same way, except perhaps `test_annotate_option`, because it needs a package with `small-fake-a` sub-dependency.

**Comment by atugushev** [comment]
Closed in #922.
